src/components/Icon/Icon.tsx

```diff
@@ -21,6 +21,10 @@ export interface IconSubComponentProps {
   "data-testid"?: string;
 }
 
+function renderIcon(Icon: SubIcon, props: IconSubComponentProps): JSX.Element {
```

Comment:
TS is usually pretty good at inferring types:

```suggestion
function renderIcon(Icon: SubIcon, props: IconSubComponentProps) {
```